### PR TITLE
python3Packages.pytorch-lightning: 1.6.5 -> 1.8.0

### PR DESCRIPTION
**Pull request by jiegec**
###### Description of changes

The new version depends on two new packages: lightning-lite and lightning-utilities.

https://github.com/Lightning-AI/lightning/releases/tag/1.8.0

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jiegec** [comment]
Result of `nixpkgs-review pr 198943` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python310Packages.lightning-lite</li>
    <li>python310Packages.lightning-utilities</li>
    <li>python310Packages.pytorch-lightning</li>
    <li>python39Packages.lightning-lite</li>
    <li>python39Packages.lightning-utilities</li>
    <li>python39Packages.pytorch-lightning</li>
  </ul>
</details>

File: pkgs/development/python-modules/pytorch-lightning/default.nix [hunk 1]
```diff
@@ -1,39 +1,28 @@
...
-    torch
-    pyyaml
-    tensorboard
+    lightning-lite
+    lightning-utilities
     torchmetrics
-    tqdm
   ];
 
   checkInputs = [ pytestCheckHook ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pytorch-lightning/default.nix:28]: The test situation did not improve?

Also nixpkgs

File: pkgs/development/python-modules/pytorch-lightning/default.nix [hunk 2]
```diff
@@ -1,39 +1,28 @@
...
   disabled = isPy27;
 
-  src = fetchFromGitHub {
-    owner = "PyTorchLightning";
-    repo = pname;
-    rev = "refs/tags/${version}";
-    hash = "sha256-CgD5g5nhz2DI4gOQyPl8/Cq6wWHzL0ALgOB5SgUOgaI=";
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-3v+bx5eOzryPReiBre9l3I2fRVTojDsGT4BYfzKrFY0=";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pytorch-lightning/default.nix:19]: Usually there is no reason to switch back, or is there one here?

File: pkgs/development/python-modules/lightning-utilities/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    owner = "Lightning-AI";
+    repo = "utilities";
+    rev = "v${version}";
+    hash = "sha256-SL27X4pykHfTFub6Lu2imZ34T8syfaRfRhfFKUaeIy8=";
+  };
+
+  propagatedBuildInputs = [
+    fire
+  ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/lightning-utilities/default.nix:21]: What about tests?

File: pkgs/development/python-modules/lightning-lite/default.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  };
+
+  propagatedBuildInputs = [
+    fsspec
+    lightning-utilities
+    numpy
+    packaging
+    torch
+  ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/lightning-lite/default.nix:27]: what about tests?

**Comment by bcdarwin** [comment]
Closing as obsolete (currently we are on version 1.9.0 on nixpkgs master).

**Pull request closed without merging**